quixstreams/kafka/configuration.py

```diff
@@ -0,0 +1,147 @@
+from typing import Literal, Optional, Callable, Tuple, List, get_args
+
+from pydantic import AliasGenerator, SecretStr, AliasChoices, Field
+from pydantic.functional_validators import BeforeValidator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from typing_extensions import Self, Annotated
+
+__all__ = ("ConnectionConfig",)
+
+MECHANISM_ALIAS = AliasChoices("sasl_mechanism", "sasl_mechanisms")
+
+
+class ConnectionConfig(BaseSettings):
+    """
+    Provides an interface for all librdkafka connection-based configs.
+
+    Allows converting to or from a librdkafka dictionary.
+
+    Also obscures secrets and handles any case sensitivity issues.
+    """
+
+    model_config = SettingsConfigDict(
+        alias_generator=AliasGenerator(
+            # used during model_dumps
+            serialization_alias=lambda field_name: field_name.replace("_", "."),
+        ),
+    )
+
+    @classmethod
+    def secret_fields(cls) -> List[str]:
```

Comment:
I mostly had it that way since arguably it doesn't need an instance to use it...but it's def only gonna be used internally by an instance method, which prob means it doesn't need to be a `classmethod` itself :+1: